src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriteStream.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A stream abstraction over the TDS protocol which works to write the TDS packet
+    /// This stream will take care of wrapping the bytes with the TDS header. 
+    /// </summary>
+    internal class TdsWriteStream : Stream, ITdsWriteStream
+    {
+        private Stream _underlyingStream;
+
+        private byte[] _WriteBuffer;
+
+        // Start at the end of the Tds header
+        internal int WriteBufferOffset { get; private set; } = TdsEnums.HEADER_LEN;
```

Comment:
I'd say this probably doesn't need to be internal and can just be a private field. While I could see the benefit of exposing it for testing, I think it can just as easily be verified by ensuring the packets written have correct byte contents.

Reply:
Oh right. I missed it. no point exposing this.